### PR TITLE
Upgrade dependencies (WDIO v9)

### PR DESCRIPTION
**Pull request by HannaTarasevich**
Fix https://github.com/webdriverio-community/wdio-cucumberjs-json-reporter/issues/582

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -56,12 +56,12 @@
...
+    "@wdio/reporter": "^9.0.0",
     "fs-extra": "^11.1.1",
     "strip-ansi": "^7.1.0"
   },
   "peerDependencies": {
-    "@wdio/cucumber-framework": "^8.0.0",
+    "@wdio/cucumber-framework": "^9.0.0",
     "@wdio/globals": "^8.0.0",
     "@wdio/types": "^8.0.0",
     "webdriverio": "^8.0.0"
```
Comment (Sven-Seyfert) [inline package.json:67]: First of all, thank you Hanna for the PR and work 👌 .

One little question:
Why did you not update the other wdio peerDependencies to v9?

File: tests/__snapshots__/metadata.spec.ts.snap [hunk 2]
```diff
@@ -58,7 +58,7 @@ exports[`metadata > determineBrowserData > should return that the browser name a
 {
   "browser": {
     "name": "configCapabilities browser name",
-    "version": "configCapabilities browser version",
+    "version": "No metadata.browser.version available",
```
Comment (Sven-Seyfert) [inline tests/__snapshots__/metadata.spec.ts.snap:61]: Are you sure about this? I mean should this really be the expected value in the test?
If so, all fine, I am just curious and a bit confused 😅 .

File: tests/__snapshots__/metadata.test.ts.snap [hunk 3]
```diff
@@ -58,7 +58,7 @@ exports[`metadata > determineBrowserData > should return that the browser name a
 {
   "browser": {
     "name": "configCapabilities browser name",
-    "version": "configCapabilities browser version",
+    "version": "No metadata.browser.version available",
```
Comment (Sven-Seyfert) [inline tests/__snapshots__/metadata.test.ts.snap:61]: Same like the comment before.

File: package.json (hunk 1 above)
Comment (HannaTarasevich) [reply package.json]: Thanks for pointing this out, updated

File: tests/__snapshots__/metadata.spec.ts.snap (hunk 2 above)
Comment (HannaTarasevich) [reply tests/__snapshots__/metadata.spec.ts.snap:61]: Right, there is ConfigCapabilities interface in use that includes browserName only, so the current logic is to determineBrowserData:
1) by current capabilities (interface includes both browserName & browserVersion)
2) if there is no "currentCapabilities" - by ConfigCapabilities (includes only browserName)
3) if  there is no 1 && 2 - use metadata info

So the test with this snapshot covers the second case

File: tests/__snapshots__/metadata.test.ts.snap (hunk 3 above)
Comment (HannaTarasevich) [reply tests/__snapshots__/metadata.test.ts.snap:61]: And the same answer

File: tests/__snapshots__/metadata.spec.ts.snap (hunk 2 above)
Comment (Sven-Seyfert) [reply tests/__snapshots__/metadata.spec.ts.snap:61]: Thanks @HannaTarasevich for the explanation. Now I got it and learned something, thanks.

File: tests/__mocks__/mocks.ts [hunk 4]
```diff
@@ -1,16 +1,16 @@
...
     DesiredCapabilitiesExtended,
     HookStatsExtended,
     RunnerStatsExtended,
     SuiteStatsExtended,
     TestStatsExtended,
-    TestrunnerExtended,
+    TestrunnerExtended
 } from '../../src/types/wdio'
 import type { Feature, Scenario, cjson_metadata } from '../../src/types'
+import { PageLoadingStrategy } from '@wdio/types/build/Capabilities'
```
Comment (christian-bromann) [inline tests/__mocks__/mocks.ts:13]: Let's not import from a package path as that can change, better from the package directly and then do `Capabilities.PageLoadingStrategy`

```suggestion
import { Capabilities } from '@wdio/types'
```

**Review by christian-bromann — CHANGES_REQUESTED** [review]

**Comment by christian-bromann** [comment]
This looks great, one last suggestion.

File: tests/__mocks__/mocks.ts (hunk 4 above)
Comment (HannaTarasevich) [reply tests/__mocks__/mocks.ts]: Sure, updated. Thanks!

**Review by christian-bromann — APPROVED** [review]
Excellent work @HannaTarasevich 👌

**Comment by christian-bromann** [comment]
I've release `v6.0.0` of this package. Let me know if this works with v9 of WebdriverIO and let's iterate on improvements if necessary. Thanks again @HannaTarasevich !

**Comment by HannaTarasevich** [comment]
@christian-bromann 
I've checked locally with an installed WebdriverIO from scratch. From my perspective, it works perfectly with v9. I raised an additional PR to improve platform and device determination.

Thanks :)

**Comment by Sven-Seyfert** [comment]
Thanks to both of you @HannaTarasevich and @christian-bromann, well done 👌 .
Now I also will update to wdio v9 _#yeah_ .

**Pull request merged**